### PR TITLE
fixes #8228 - add a concern to serve config_templates controllers

### PR DESCRIPTION
**Pull request by shlomizadok**
...

### CODE REVIEW

File: app/controllers/concerns/foreman/controller/config_templates_mixin.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  private
+
+  def default_template_url(template, hostgroup)
+    url_for :host => Setting[:unattended_url], :action => :template, :controller => '/unattended',
+            :id => template.name, :hostgroup => hostgroup.name
+  end
+
+  # convert the file upload into a simple string to save in our db.
+  def handle_template_upload
+    return unless params[:config_template] && (t=params[:config_template][:template])
```
Comment (dLobatog) [inline app/controllers/concerns/foreman/controller/config_templates_mixin.rb]: Please use meaningful variables, t -> template, tk -> template_kind.

File: app/controllers/concerns/foreman/controller/config_templates_mixin.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module Foreman::Controller::ConfigTemplatesMixin
```
Comment (dLobatog) [inline app/controllers/concerns/foreman/controller/config_templates_mixin.rb]: Don't use `Mixin` in the module name, it's very redundant (something like "class UserModel" for app/models/user.rb), maybe there's some word that defines better the purpose of this module.

File: app/controllers/concerns/foreman/controller/config_templates_mixin.rb (hunk 1 above)
Comment (shlomizadok) [inline app/controllers/concerns/foreman/controller/config_templates_mixin.rb]: For the record, I was just copying shared methods from the controllers and put them here. But I completely agree it should have meaningful names. refactored and pushing in 3, 2, 1...

**Comment by dLobatog** [comment]
Looks pretty good :+1: to merge if tests are green

**Comment by dLobatog** [comment]
Merged as 563fa283aff363801d5cd160ab0c54f6c7634d98, thanks @shlomizadok !

**Pull request closed without merging**